File: 04_Frontend_Application/Edums-Frontend/src/app/modules/m1-school-office/interface/school-audit-log.ts
import type { OperationType, SeverityLevel } from './common';

export interface SchoolAuditLog {
  id: number;
  schoolId: number;
  affectedTableName: string;
  affectedEntityId: number;
  operationType: OperationType;
  changeTypeSummary: string;
  oldValueJson: string | null;
  newValueJson: string | null;
  changeSummaryText: string;
  performedByUserId: number;
  performedByUserName: string;
  performedByUserRole: string;
  ipAddress: string | null;
  deviceInfo: string | null;
  actionDate: string;
  severityLevel: SeverityLevel;
  isSuspicious: boolean;
  decisionDocumentUrl: string | null;
  notes: string | null;
}

export type CreateSchoolAuditLogDto = Omit<SchoolAuditLog, 'id'>;

export type UpdateSchoolAuditLogDto = Omit<SchoolAuditLog, 'schoolId'>;

File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/SchoolAuditLog.cs
using System;
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

/// <summary>
/// سجل التغييرات والتدقيق المدرسي - School-level audit trail extracted from ZIP ERD SchoolAuditLog table (lines 304-326).
/// Records every Add/Edit/Delete event on critical school entities with user, IP and severity tracking.
/// </summary>
public class SchoolAuditLog : BaseAuditableEntity
{
    public long SchoolId { get; set; }
    public string AffectedTableName { get; set; } = string.Empty;  // e.g. "Students", "Employees"
    public long AffectedEntityId { get; set; }
    public int OperationType { get; set; } // 1=Insert, 2=Update, 3=Delete, 4=Restore
    public string ChangeTypeSummary { get; set; } = string.Empty;   // Human-readable change label
    public string? OldValueJson { get; set; }
    public string? NewValueJson { get; set; }
    public string ChangeSummaryText { get; set; } = string.Empty;
    public long PerformedByUserId { get; set; }
    public string PerformedByUserName { get; set; } = string.Empty;
    public string PerformedByUserRole { get; set; } = string.Empty;
    public string? IpAddress { get; set; }
    public string? DeviceInfo { get; set; }
    public DateTime ActionDate { get; set; } = DateTime.UtcNow;
    public int SeverityLevel { get; set; } = 1; // 1=Info, 2=Warning, 3=Critical
    public bool IsSuspicious { get; set; }
    public string? DecisionDocumentUrl { get; set; }
    public string? Notes { get; set; }

    // Navigation Property
    public virtual School? School { get; set; }
}
